sklearn/pipeline.py

```diff
@@ -350,7 +356,8 @@ def fit(self, X, y=None, **fit_params):
                 self._final_estimator.fit(Xt, y, **fit_params)
         return self
 
-    def fit_transform(self, X, y=None, **fit_params):
```

Comment:
We never did this before? This will need a small change log entry.

Reply:
Saying that ``fit_transform`` now does proper ducktyping?

Reply:
Should this be a separate PR?

Reply:
> Saying that fit_transform now does proper ducktyping?

I don't much mind

Reply:
I'm not sure what you mean.